src/lib/ppx_coda/versioned.ml

```diff
@@ -96,19 +239,29 @@ let versioned_syntactic_check =
             let results =
               List.map structure ~f:(fun si -> self#structure_item si new_acc)
             in
-            let _new_pmb_expr, _modules = List.unzip results in
-            (* TODO : do the deriving *)
-            let new_str =
-              {pstr_desc= Pstr_module module_details; pstr_loc= str.pstr_loc}
+            let new_str, _accs = List.unzip results in
+            let new_module =
+              { str with
+                pstr_desc=
+                  Pstr_module
+                    { module_details with
+                      pmb_expr=
+                        {pmb_expr with pmod_desc= Pmod_structure new_str} } }
             in
-            (new_str, acc)
+            (new_module, acc)
         | _ -> (str, acc) )
-      (* for type declaration, check validity when versioned *)
-      | Pstr_type (_rec_decl, type_decls) ->
+      | Pstr_type (_rec_decl, type_decls) as type_decl ->
+          (* if versioned, check syntactic placement of type, add code *)
           let inner3_modules = List.take acc 3 in
-          List.iter type_decls ~f:(fun ty_decl ->
-              validate_versioned ty_decl inner3_modules ) ;
-          (str, acc)
+          (* list of generated declarations for versioned types *)
+          let version_decls =
+            List.fold type_decls ~init:[] ~f:(fun decls ty_decl ->
+                validate_and_generate_versioned decls ty_decl inner3_modules )
+          in
+          if List.is_empty version_decls then (str, acc)
+          else
+            ( add_version_decls_to_type str type_decl version_decls pstr_loc
```

Comment:
Since we're doing the deriving here, you'll need to take `version` out of the deriving list, otherwise `ppx_deriving` will complain

Reply:
You mean in the transformed code?

Reply:
Sadly yes. We can't throw out the deriving completely in case there are other things to be derived (I think), but we'll get an error if we leave it in..